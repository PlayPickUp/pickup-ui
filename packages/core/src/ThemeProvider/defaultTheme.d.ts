import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
