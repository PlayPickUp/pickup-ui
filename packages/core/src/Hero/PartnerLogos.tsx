import { DefaultTheme, PartnerLogoProps } from "../types";
import React from "react";
import { createUseStyles } from "react-jss";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  logoList: {
    width: "100%",
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "center",
    alignContent: "center",
    marginBottom: theme.spacing.base * 2,
  },
  logo: {
    flex: "3 0 26%",
    margin: [0, theme.spacing.base * 4, theme.spacing.base * 2],
    display: "flex",
    justifyContent: "center",
    [theme.mediaQuery(theme.breakpoints.small)]: {
      margin: [theme.spacing.base, theme.spacing.base * 8],
      flex: "8 0 12%",
    },
  },
  logoImage: {
    width: "100%",
    maxWidth: theme.spacing.base * 37,
  },
}));

const PartnerLogos: React.FC<PartnerLogoProps> = ({ logos }) => {
  const classes = useStyles();
  if (!logos?.length) return null;
  return (
    <div className={classes.logoList}>
      {logos.map((logo, index) => (
        <div key={index} className={classes.logo}>
          <img src={logo.url} alt={logo.alt} className={classes.logoImage} />
        </div>
      ))}
    </div>
  );
};

export default PartnerLogos;
