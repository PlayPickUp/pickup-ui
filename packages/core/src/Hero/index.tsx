import React, { useState, useEffect } from "react";
import { createUseStyles } from "react-jss";
import { DefaultTheme, HeroProps, HeroStyleState } from "../types";
import Typography from "../Typography";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  root: {
    backgroundColor: theme.colors.grey.light,
    paddingTop: theme.spacing.base * 8,
    paddingBottom: theme.spacing.base * 8,
    textAlign: "center",
    [theme.mediaQuery(theme.breakpoints.small)]: {
      textAlign: "left",
    },
  },
  container: {
    paddingLeft: theme.spacing.base * 7,
    paddingRight: theme.spacing.base * 7,
    maxWidth: theme.spacing.base * 214,
  },
  row: {
    display: "flex",
    flexDirection: "column",
    [theme.mediaQuery(theme.breakpoints.small)]: {
      flexDirection: "row-reverse",
      height: "100%",
    },
  },
  column: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    "& > div, & > h2, & > h3, & > p, & > img": {
      paddingTop: theme.spacing.base * 2,
      paddingBottom: theme.spacing.base * 2,
    },
    [theme.mediaQuery(theme.breakpoints.small)]: {
      padding: 0,
      justifyContent: "left",
    },
  },
  imageColumn: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    width: "100%",
    paddingBottom: theme.spacing.base * 2,
    [theme.mediaQuery(theme.breakpoints.small)]: {
      paddingBottom: 0,
    },
  },
  eyebrow: {
    display: "flex",
    flexDirection: "row",
    "& > :first-child": {
      paddingRight: theme.spacing.base * 2,
      color: theme.colors.purple.base,
    },
    [theme.mediaQuery(theme.breakpoints.small)]: {
      width: "100%",
    },
  },
  title: {
    fontWeight: "normal",
    width: "100%",
    [theme.mediaQuery(theme.breakpoints.small)]: {
      fontSize: 34,
      lineHeight: `${theme.spacing.base * 10}px`,
    },
  },
  description: {
    fontSize: 13,
    width: "100%",
    lineHeight: `${theme.spacing.base * 5}px`,
    [theme.mediaQuery(theme.breakpoints.small)]: {
      fontSize: 18,
      lineHeight: `${theme.spacing.base * 7}px`,
    },
  },
  image: {
    maxWidth: 300,
    width: "100%",
    height: "auto",
    borderRadius: 15,
  },
}));

const Hero: React.FC<HeroProps> = ({
  title,
  description,
  image_src,
  image_alt,
  eyebrow,
}) => {
  const classes = useStyles();
  const [heroStyle, setHeroStyle] = useState<HeroStyleState>({
    titleVariant: "heading3",
    descriptionVariant: "body",
  });

  useEffect(() => {
    if (eyebrow) {
      setHeroStyle({
        ...heroStyle,
        titleVariant: "heading2",
      });
    }
  }, []);

  return (
    <div className={classes.root}>
      <div className={classes.container}>
        <div className={classes.row}>
          <div className={classes.imageColumn}>
            <img className={classes.image} src={image_src} alt={image_alt} />
          </div>
          <div className={classes.column}>
            {eyebrow ? (
              <div className={classes.eyebrow}>
                <Typography variant="body2">{eyebrow.name}</Typography>
                <Typography variant="body2">{eyebrow.description}</Typography>
              </div>
            ) : null}
            <Typography
              data-testid="hero-title"
              className={classes.title}
              variant={heroStyle.titleVariant}
            >
              {title}
            </Typography>
            <Typography
              data-testid="hero-description"
              className={classes.description}
              variant={heroStyle.descriptionVariant}
            >
              {description}
            </Typography>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Hero;
