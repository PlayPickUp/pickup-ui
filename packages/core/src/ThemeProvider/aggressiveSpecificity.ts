/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
