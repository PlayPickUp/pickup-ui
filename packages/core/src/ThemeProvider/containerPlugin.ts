/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
