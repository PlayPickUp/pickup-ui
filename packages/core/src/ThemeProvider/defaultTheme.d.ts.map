{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}