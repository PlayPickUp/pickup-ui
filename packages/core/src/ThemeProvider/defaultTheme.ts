import {
  Breakpoints,
  CreateShadow,
  DefaultGradients,
  DefaultShadows,
  DefaultSpacing,
  DefaultTheme,
  DefaultThemeColors,
  DefaultTypography,
  DefaultZIndex,
  FontFamilies,
  GreenMap,
  GreyMap,
  MediaQuery,
  PurpleMap,
  RedMap,
  ResponsiveFonts,
} from "../types";

// Color me PickUp âš¡
const purple: PurpleMap = {
  light: "#F5F0FF",
  lightBase: "#AFA0FF",
  base: "#7A60FF",
  dark: "#5E24FF",
  transparent: "#EAE5FF",
};

const green: GreenMap = {
  light: "#EAFDF7",
  base: "#25DA95",
};

const grey: GreyMap = {
  light: "#F3F2F5",
  lightBase: "#DDDDDD",
  base: "#9D9D9E",
  dark: "#615E66",
  medium: "#DFDAE6",
};

const red: RedMap = {
  light: "#FFF1F4",
  base: "#FF1244",
};

const colors: DefaultThemeColors = {
  primary: { ...purple },
  secondary: { ...grey },
  green,
  purple,
  grey,
  red,
  black: "#2C2933",
  white: "#FFF",
  success: green.base,
  error: red.base,
};

// Font and Typography
const fontFamilies: FontFamilies = {
  body: "'Inter', Helvetica, sans-serif",
  headline: "'Ringside-Condensed', Helvetica, sans-serif",
};

const fontStyles: ResponsiveFonts = {
  desktop: {
    title: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 48,
      lineHeight: "54px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading2: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 36,
      lineHeight: "40px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading3: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 28,
      lineHeight: "34px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading4: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 22,
      lineHeight: "28px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading5: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 16,
      lineHeight: "22px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading6: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 13,
      lineHeight: "16px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 18,
      lineHeight: "30px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body2: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 13,
      lineHeight: "20px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
  },
  mobile: {
    title: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 34,
      lineHeight: "38px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading2: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 28,
      lineHeight: "34px",
      letterSpacing: "0.1px",
      textTransform: "uppercase",
      color: "inherit",
    },
    heading3: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 22,
      lineHeight: "26px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading4: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 16,
      lineHeight: "18px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading5: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 13,
      lineHeight: "15px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    heading6: {
      fontFamily: fontFamilies.headline,
      fontWeight: 600,
      fontSize: 11,
      lineHeight: "13px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 15,
      lineHeight: "24px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
    body2: {
      fontFamily: fontFamilies.body,
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: 12,
      lineHeight: "18px",
      letterSpacing: "0.1px",
      color: "inherit",
    },
  },
};

const typography: DefaultTypography = {
  fontFamilies,
  fontStyles,
};

// Fancy ass Material UI inspired shadows
const shadowKeyUmbraOpacity = 0.2;
const shadowKeyPenumbraOpacity = 0.14;
const shadowAmbientShadowOpacity = 0.12;

const createShadow: CreateShadow = (...px) => {
  return [
    `${px[0]}px ${px[1]}px ${px[2]}px ${px[3]}px rgba(0, 0, 0, ${shadowKeyUmbraOpacity})`,
    `${px[4]}px ${px[5]}px ${px[6]}px ${px[7]}px rgba(0, 0, 0, ${shadowKeyPenumbraOpacity})`,
    `${px[8]}px ${px[9]}px ${px[10]}px ${px[11]}px rgba(0, 0, 0, ${shadowAmbientShadowOpacity})`,
  ].join(",");
};

const shadows: DefaultShadows = [
  createShadow(0, 1, 3, 0, 0, 1, 1, 0, 0, 2, 1, -1),
  createShadow(0, 1, 5, 0, 0, 2, 2, 0, 0, 3, 1, -2),
  createShadow(0, 1, 8, 0, 0, 3, 4, 0, 0, 3, 3, -2),
  createShadow(0, 2, 4, -1, 0, 4, 5, 0, 0, 1, 10, 0),
  createShadow(0, 3, 5, -1, 0, 5, 8, 0, 0, 1, 14, 0),
  createShadow(0, 3, 5, -1, 0, 6, 10, 0, 0, 1, 18, 0),
  createShadow(0, 4, 5, -2, 0, 7, 10, 1, 0, 2, 16, 1),
  createShadow(0, 5, 5, -3, 0, 8, 10, 1, 0, 3, 14, 2),
  createShadow(0, 5, 6, -3, 0, 9, 12, 1, 0, 3, 16, 2),
  createShadow(0, 6, 6, -3, 0, 10, 14, 1, 0, 4, 18, 3),
  createShadow(0, 6, 7, -4, 0, 11, 15, 1, 0, 4, 20, 3),
  createShadow(0, 7, 8, -4, 0, 12, 17, 2, 0, 5, 22, 4),
  createShadow(0, 7, 8, -4, 0, 13, 19, 2, 0, 5, 24, 4),
  createShadow(0, 7, 9, -4, 0, 14, 21, 2, 0, 5, 26, 4),
  createShadow(0, 8, 9, -5, 0, 15, 22, 2, 0, 6, 28, 5),
  createShadow(0, 8, 10, -5, 0, 16, 24, 2, 0, 6, 30, 5),
  createShadow(0, 8, 11, -5, 0, 17, 26, 2, 0, 6, 32, 5),
  createShadow(0, 9, 11, -5, 0, 18, 28, 2, 0, 7, 34, 6),
  createShadow(0, 9, 12, -6, 0, 19, 29, 2, 0, 7, 36, 6),
  createShadow(0, 10, 13, -6, 0, 20, 31, 3, 0, 8, 38, 7),
  createShadow(0, 10, 13, -6, 0, 21, 33, 3, 0, 8, 40, 7),
  createShadow(0, 10, 14, -6, 0, 22, 35, 3, 0, 8, 42, 7),
  createShadow(0, 11, 14, -7, 0, 23, 36, 3, 0, 9, 44, 8),
  createShadow(0, 11, 15, -7, 0, 24, 38, 3, 0, 9, 46, 8),
];

// Base Unit Spacings
const spacing: DefaultSpacing = {
  base: 4, // ðŸš¨ changing this value (4) could have VERY BAD CONCEQUENCES! BE VERY SURE, MAYBE YOU JUST NEED TO ADD A VALUE? ðŸ¤”
};

// Default zIndex values
const zIndex: DefaultZIndex = {
  overlay: 999,
};

// Default Border Radius
const borderRadius = 4;

// Media Queries
const mediaQuery: MediaQuery = (minWidth: number) => {
  return `@media (min-width: ${minWidth}px)`;
};

// Breakpoints
const breakpoints: Breakpoints = {
  small: 787,
  medium: 1029,
  large: 1280,
};

// Default Gradient Values
const gradients: DefaultGradients = {
  main: `linear-gradient(112.25deg, #914EFF -20.38%, #5D1BFF 122.56%)`,
  hover: `linear-gradient(112.25deg, #914EFF 35.38%, #5D1BFF 122.56%)`,
  disabled:
    "linear-gradient(112.25deg, rgba(145,78,255,0.2) -20.38%, rgba(93,27,255,0.2) 122.56%)",
};

export const defaultTheme: DefaultTheme = {
  borderRadius,
  breakpoints,
  colors,
  gradients,
  mediaQuery,
  shadows,
  spacing,
  typography,
  zIndex,
};
