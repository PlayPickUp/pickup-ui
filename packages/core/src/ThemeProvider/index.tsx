import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
