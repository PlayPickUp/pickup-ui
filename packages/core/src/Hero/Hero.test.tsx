import React from "react";
import { render, within } from "@testing-library/react";

import { BrowserRouter as Router } from "react-router-dom";

import ThemeProvider from "../ThemeProvider";
import Hero from "./index";

const breadCrumbs = [
  {
    name: "Crumb 1",
    path: "/crumb1",
  },
  {
    name: "Crumb 2",
    path: "/crumb2",
  },
];

test("Renders and matches snapshot", () => {
  const { container } = render(
    <ThemeProvider>
      <Hero
        title="Hero Title"
        description="Hero Description"
        image_src="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/bowlero_lanes.jpg"
        image_alt="bowlero"
        eyebrow={{ name: "Eyebrow Title", description: "Eyebrow Description" }}
      />
    </ThemeProvider>
  );
  expect(container).toMatchSnapshot();
});

test("Passes and renders props", () => {
  const { getByText, getByAltText, getByRole } = render(
    <ThemeProvider>
      <Router>
        <Hero
          title="Hero Title"
          description="Hero Description"
          image_src="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/bowlero_lanes.jpg"
          image_alt="bowlero"
          eyebrow={{
            name: "Eyebrow Title",
            description: "Eyebrow Description",
          }}
          chip="Chip"
          crumbs={breadCrumbs}
        />
      </Router>
    </ThemeProvider>
  );
  // title
  expect(getByText("Hero Title")).toBeTruthy();
  // description
  expect(getByText("Hero Description")).toBeTruthy();
  // image_src and image_alt
  expect(getByAltText("bowlero").getAttribute("src")).toEqual(
    "https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/bowlero_lanes.jpg"
  );
  // eyebrow
  expect(getByText("Eyebrow Title | Eyebrow Description")).toBeTruthy();
  // chip
  expect(getByText("Chip")).toBeTruthy();
  // crumbs
  const list = getByRole("list", { name: /breadcrumbs/i });
  const { getAllByRole } = within(list);
  const items = getAllByRole("listitem");
  const breadcrumbNames = items.map((item) => item.textContent);
  expect(breadcrumbNames).toMatchInlineSnapshot(`
    Array [
      "Crumb 1/",
      "Crumb 2",
    ]
  `);
});
