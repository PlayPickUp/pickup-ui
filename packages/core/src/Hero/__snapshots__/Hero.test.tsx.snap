// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders and matches snapshot 1`] = `
body #PickUpUI .j7 {
  margin: 0;
  padding: 0;
  position: relative;
  vertical-align: baseline;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body #PickUpUI .j8 {
  color: inherit;
}

body #PickUpUI .j9 {
  color: inherit;
  font-size: 28px;
  font-family: 'Ringside-Condensed', Helvetica, sans-serif;
  font-weight: 600;
  line-height: 34px;
  letter-spacing: 0.1px;
  text-transform: uppercase;
}

body #PickUpUI .j12 {
  color: inherit;
}

body #PickUpUI .j13 {
  color: inherit;
  font-size: 12px;
  font-style: normal;
  font-family: 'Inter', Helvetica, sans-serif;
  font-weight: normal;
  line-height: 18px;
  letter-spacing: 0.1px;
}

body #PickUpUI .j14 {
  color: inherit;
}

body #PickUpUI .j15 {
  color: inherit;
  font-size: 15px;
  font-style: normal;
  font-family: 'Inter', Helvetica, sans-serif;
  font-weight: normal;
  line-height: 24px;
  letter-spacing: 0.1px;
}

body #PickUpUI .j20 {
  width: 100%;
  margin: 8px 0;
  display: flex;
  flex-wrap: wrap;
  align-content: center;
  justify-content: center;
}

body #PickUpUI .j21 {
  flex: 3 0 26%;
  margin: 4px 16px;
  display: flex;
  justify-content: center;
}

body #PickUpUI .j22 {
  width: 100%;
  max-width: 148px;
}

body #PickUpUI .j17 {
  background: #261733;
  color: #FFF;
  border: none;
  height: auto;
  margin: 0px 0px 0px 0px;
  display: flex;
  opacity: 1;
  padding: 0 30px;
  position: relative;
  flex-flow: row nowrap;
  font-size: 18px;
  appearance: none;
  background: linear-gradient(112.25deg, #914EFF -20.38%, #5D1BFF 122.56%);
  min-height: 47px;
  text-align: center;
  transition: opacity 200ms ease-in-out;
  align-items: center;
  font-family: 'Inter', Helvetica, sans-serif;
  line-height: 24px;
  border-radius: 4px;
  letter-spacing: 0.3px;
  text-transform: capitalize;
  justify-content: center;
}

body #PickUpUI .j18 {
  width: auto;
}

body #PickUpUI .j0 {
  color: #2C2933;
  width: 100%;
  text-align: center;
  background-color: #F6F4FF;
}

body #PickUpUI .j1 {
  margin: 0 auto;
  padding: 20px 20px;
  box-sizing: border-box;
}

body #PickUpUI .j2 {
  display: flex;
  flex-direction: column;
}

body #PickUpUI .j3 {
  width: 100%;
  display: flex;
  align-items: center;
  padding-bottom: 8px;
  justify-content: center;
}

body #PickUpUI .j4 {
  height: 100%;
  overflow: hidden;
  border-radius: 4px;
}

body #PickUpUI .j5 {
  height: auto;
  display: block;
  max-width: 100%;
  object-fit: cover;
}

body #PickUpUI .j6 {
  width: 100%;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}

body #PickUpUI .j10 {
  width: 100%;
  font-size: 30px;
  font-weight: normal;
  line-height: 36px;
}

body #PickUpUI .j11 {
  display: flex;
  flex-direction: row;
}

body #PickUpUI .j16 {
  width: 100%;
  font-size: 13px;
  line-height: 20px;
}

body #PickUpUI .j19 {
  width: 240px;
  font-size: 16px;
  margin-top: 12px;
  font-family: 'Ringside-Condensed', Helvetica, sans-serif;
  font-weight: normal;
  letter-spacing: 1px;
  text-transform: uppercase;
  text-decoration: none;
}

<div>
  <div
    id="PickUpUI"
  >
    <div
      class="j0"
    >
      <div
        class="j1"
      >
        <div
          class="j2"
        >
          <div
            class="j3"
          >
            <div
              class="j4"
            >
              <img
                alt="bowlero"
                class="j5"
                src="https://playpickup.s3.us-east-2.amazonaws.com/away-team/kasper/homebase/prize-images/bowlero_lanes.jpg"
              />
            </div>
          </div>
          <div
            class="j6"
          >
            <h2
              class="j7 j8 j9 j10"
              data-testid="hero-title"
            >
              Hero Title
            </h2>
            <div
              class="j11"
            >
              <p
                class="j7 j12 j13"
              >
                Eyebrow Title | Eyebrow Description
              </p>
            </div>
            <p
              class="j7 j14 j15 j16"
              data-testid="hero-description"
            >
              Hero Description
            </p>
            <a
              class="j17 j18 j19"
              data-testid="pickup-button"
              href="https://www.playpickup.com"
            >
              CTA Button
            </a>
          </div>
        </div>
        <div
          class="j20"
        >
          <div
            class="j21"
          >
            <img
              alt="Image 1"
              class="j22"
              src="https://picsum.photos/125/60"
            />
          </div>
          <div
            class="j21"
          >
            <img
              alt="Image 2"
              class="j22"
              src="https://picsum.photos/125/60"
            />
          </div>
          <div
            class="j21"
          >
            <img
              alt="Image 3"
              class="j22"
              src="https://picsum.photos/125/60"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
